test/auth_test.js

```diff
@@ -309,6 +309,58 @@ suite('api/auth', function() {
     ],
   });
 
+  testEndpoint({
+    method: 'get',
+    route: '/test-id-scope',
+    name: 'test-id-scope',
+    scopes: {AllOf: [{for: 'scope', in: 'scopes', each: '<scope>'}]},
+    handler: async (req, res) => {
+      res.status(200).json({
+        clientId: await req.clientId(),
+      });
+    },
+    tests : [
+      {
+        label: 'client has sufficient scopes',
+        id: 'admin',
+        tester: (auth, url) => request.get(url).hawk(auth)
+          .then(function(res) {
+            assert(res.body.clientId == 'admin');
+            return res;
+          }),
+      },
+    ],
+  });
+
+  testEndpoint({
+    method: 'get',
+    route: '/test-scope',
+    name: 'test-scope',
+    scopes: {AllOf: ['service:<param>']},
+    handler: async (req, res) => {
+      try {
+        await req.authorize({param: 'myfolder/other-resource'});
+        res.reply({});
+      } catch (err) {
+        if (err.code === 'AuthorizationError') {
+          assert(err.messageTemplate.indexOf('Client ID nobody does not have sufficient scopes')!==-1);
+          return res.reportError('InsufficientScopes', err.messageTemplate, err.details);
+        }
```

Comment:
This tests the right thing, in that without your change to `auth.js` the test will fail.  But, it runs in the "server" (not really a server since it's a test and everything's on the same computer, but pretend..), rather than on the client.  Still, checking it on the "client" would require making bigger changes to how `testEndpoint` is implemented, which would be a much bigger change.  So this is good :)